test/include/rev-printf.h

```diff
@@ -210,153 +266,41 @@ static void vprintfmt( void ( *putch )( int, void** ), void** putdat, const char
     unsigned_number:
       num = getuint( &ap, lflag );
     signed_number:
-      printnum( putch, putdat, num, base, width, padc );
+      bytes += printnum( putch, putdat, num, base, width, padc );
       break;
 
     // escaped '%' character
-    case '%': putch( ch, putdat ); break;
+    case '%': {
+      putch( ch, putdat );
+      bytes++;
+      break;
+    }
 
     // unrecognized escape sequence - just print it literally
     default:
       putch( '%', putdat );
+      bytes++;
       fmt = last_fmt;
       break;
     }
   }
+  return -1;
 }
 
-int printf( const char* fmt, ... ) {
+int rev_printf( const char* fmt, ... ) {
```

Comment:
You might want to add `__attribute((format(printf, 1, 2)))` or the like, so that warnings are produced when arguments do not match formats. I seem to recall adding `-Wsuggest-attribute=format` at one point.